aws/onboarding/byok8s/python/rubrik_eks_cluster_private.py

```diff
@@ -0,0 +1,1203 @@
+            user_data: str,
+            instance_profile_arn: str,
+            security_group: str,
+            instance_metadata_options: dict,
+            key_name: Optional[str] = None
+    ) -> str:
+        """Creates launch template for worker nodes
+
+         Args:
+             name: launch template name
+             block_device_mappings: block device mapping for worker nodes
+             ami_id: AMI ID for worker nodes
+             node_type: node type
+             user_data: user data for instance
+             instance_profile_arn: The arn of the instance profile
+             instance_metadata_options: The metadata options for the instance
+             security_group: worker node security group ID
+             key_name: name of the key pair
+         Returns:
+             launch template id
+         """
+        print(f"Creating launch template {name}")
+        launch_template_data = {
+            'BlockDeviceMappings': block_device_mappings,
+            'ImageId': ami_id,
+            'InstanceType': node_type,
+            'UserData': user_data,
+            'SecurityGroupIds': [security_group],
+            'IamInstanceProfile': {
+                'Arn': instance_profile_arn
+            },
+            'MetadataOptions': instance_metadata_options
+        }
+        if key_name:
+            launch_template_data['KeyName'] = key_name
+        try:
+            resp = self.__ec2_client.create_launch_template(
+                LaunchTemplateName=name,
+                LaunchTemplateData=launch_template_data,
+            )
+            print(f"Created launch template {name}: {resp}")
+            return str(resp['LaunchTemplate']['LaunchTemplateId'])
+        except botocore.exceptions.ClientError as e:
+            print(f"Error creating launch template: {str(e)}")
+            raise
+
+    def create_autoscaling_group(
+            self,
+            name: str,
+            launch_template_id: str,
+            min_size: int,
+            max_size: int,
+            desired_capacity: int,
+            vpc_zone_identifier: str,
+            cluster_name: str
+    ) -> None:
+        """Creates autoscaling group
+
+Args:
+    name: launch template name
+    launch_template_id: launch template id
+    min_size: the minimum size of the group
+    max_size: the maximum size of the group
+    desired_capacity: initial capacity of autoscaling group at the time of creation
+    vpc_zone_identifier: A comma-separated list of subnet IDs for a virtual private cloud (VPC)
+    where instances in the Auto Scaling group can be created.
+    cluster_name: Name of the EKS cluster
+Returns:
+    None
+"""
+        print(f"Creating autoscaling group {name}")
+        try:
+            self.__autoscaling_client.create_auto_scaling_group(
+                AutoScalingGroupName=name,
+                LaunchTemplate={
+                    'LaunchTemplateId': launch_template_id,
+                    'Version': '$Latest'
```

Comment:
I think in non-byok version used is default, can you please check this once

Reply:
This is set as Latest in the byok E2E:
![Screenshot 2024-07-07 at 11 45 13 AM](https://github.com/rubrikinc/rubrik-cloud-native-protection/assets/167168896/183ea3b2-006e-48ab-935a-04cfd643beea)
and its set as Default in case of Rubrik managed cluster:
![Screenshot 2024-07-07 at 11 38 20 AM](https://github.com/rubrikinc/rubrik-cloud-native-protection/assets/167168896/736c147b-cc83-47b6-a3a8-cca4c5475e28)

Reply:
For now i have updated to Default as done in Rubrik created cluster:
![Screenshot 2024-07-08 at 9 47 38 AM](https://github.com/rubrikinc/rubrik-cloud-native-protection/assets/167168896/1984b121-614d-4f84-b97b-e7164f63d874)

Reply:
In RSC, we launch a cluster and tear it down after use, so latest == default? Over here, since the cluster is customer managed, and if a customer is changing the launch template, they'd want their changes to be reflected, so let's use `latest`?

Reply:
Make sense, Updated to latest